tests/test_template_types.py

```diff
@@ -31,54 +30,94 @@
 
 
 @pytest.mark.parametrize(
-    "template_class, expected_error",
+    "template_class, expected_errors",
     (
-        pytest.param(
+        (
             Template,
-            ("Can't instantiate abstract class Template with abstract methods __str__"),
-            marks=pytest.mark.skipif(sys.version_info >= (3, 9), reason="‘methods’ will be singular"),
+            (
+                "Can't instantiate abstract class Template with abstract method __str__",
+                (
+                    "Can't instantiate abstract class Template without an implementation "
+                    "for abstract method '__str__'"
+                ),
+            ),
```

Comment:
i dont love this - i think this makes the test a bit messy and makes harder to spot and tidy up when we've completed our migration

we could just have an if statement in the code like we do in the SerialisedModel tests

```
def test_abstract_classes_cant_be_instantiated(template_class, expected_3_10_error, expected_3_9_error):
    with pytest.raises(TypeError) as error:
        template_class({})
    if sys.version_info < (3, 10):
        assert str(error.value) == expected_3_9_error
    else:
        assert str(error.value) == expected_3_10_error

Reply:
or just super simple:

```
assert f"Can't instantiate abstract class {template_class}" in str(error.value) 
assert f"__str__" in str(error.value) 
```

(this is more in line with our guidance we're publishing to users about error messages anyway)